spec/models/reservation_spec.rb

```diff
@@ -538,4 +538,32 @@
       expect(FactoryGirl.create(:valid_reservation).approved?).to be_falsey
     end
   end
+
+  context '#late_fee' do
+    let(:reservation) do
+      r = FactoryGirl.build(:overdue_reservation)
+      r.save(validate: false)
+      r
+    end
+
+    it 'returns the correct late fee' do
+      expected = (Time.zone.today - reservation.due_date) *
+                 reservation.equipment_model.late_fee
+      expect(reservation.late_fee).to eq(expected)
+    end
+
+    it 'returns 0 if not overdue' do
+      reservation.update_attributes(due_date: Time.zone.today + 1.day,
+                                    overdue: false)
+      expect(reservation.late_fee).to eq(0)
+    end
+
+    it 'returns the cap if set' do
```

Comment:
I'm pretty sure I meant "if a late fee cap is set", but I can update the spec language to be clearer.